sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -32,138 +32,60 @@
 
 from azure.core.pipeline.policies import HttpLoggingPolicy
 
+from .http_constants import HttpHeaders
 
 
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs) -> None: # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
         if self._enable_diagnostics_logging:
-            self.logger = logger
-
-    def on_request(self, request): # pylint: disable=too-many-return-statements, too-many-statements
-        # type: (PipelineRequest) -> None
-        http_request = request.http_request
-        options = request.context.options
-        self._enable_diagnostics_logging = request.context.setdefault(
-            "enable_diagnostics_logging",
-            options.pop("enable_diagnostics_logging", self._enable_diagnostics_logging))
-        if self._enable_diagnostics_logging:
-            # Get logger in my context first (request has been retried)
-            # then read from kwargs (pop if that's the case)
-            # then use my instance logger
-            logger = request.context.setdefault("logger", options.pop("logger", self.logger))
-
+            cosmos_disallow_list = ["Authorization"]
+            cosmos_allow_list = [
+                v for k, v in HttpHeaders.__dict__.items() if not k.startswith("_") and k not in cosmos_disallow_list
+            ]
+            self.allowed_header_names.update(set(cosmos_allow_list))
```

Comment:
This logic is adding _all_ the headers in the HttpHeaders constants (obviously excluding Auth) - we should probably scrutinize that list to make sure there aren't any others that are sensitive.
This feature is gated behind an opt-in, but it's still logged at `INFO` level, and so customers should be aware of the extent of data that may be logged when they use this and they should know whether they need to handle this with a high degree of security.